test/VSSDK.TestExtension/MEF/TextviewCreationListener.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using Community.VisualStudio.Toolkit.Shared;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace TestExtension.MEF
+{
+    [Export(typeof(IWpfTextViewCreationListener))]
+    [ContentType("text")]
+    [TextViewRole(PredefinedTextViewRoles.PrimaryDocument)]
+    internal class TextviewCreationListener : IWpfTextViewCreationListener, IDisposable
+    {
+        private readonly ToolkitThreadHelper _threadHelper = new ToolkitThreadHelper(ThreadHelper.JoinableTaskContext);
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _threadHelper.Dispose();
+            }
+        }
+
+        public void TextViewCreated(IWpfTextView textView)
+        {
+            // Note: RunAsync is not awaited/joined
+            _threadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                try
+                {
+                    await _threadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(_threadHelper.DisposalToken);
+
+                    // Do work, e.g.
+                    // await MyMethodAsync(_threadHelper.DisposalToken);
+                }
+                catch (Exception ex)
+                {
+                    await ex.LogAsync();
+                }
```

Comment:
Doesn't the toolkit already have a `LogAndForget` extension method that can be used on the result of `RunAsync` so it doesn't have to be done here, if this is all that the catch block will contain?

Reply:
It has this extension method but only implemented for Task, not JoinableTask:

```
public static void ForgetAndLogOnFailure(this System.Threading.Tasks.Task task)
{
    task.ContinueWith(delegate (System.Threading.Tasks.Task antecedent)
    {
        antecedent.Exception!.Log();

    }, CancellationToken.None, TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default).Forget();
}
```

Would an equivalent extension method for JoinableTask just need to do the same with the `JoinableTask.Task`?

Reply:
Yes, that would do it. You could also do it without an additional extension method like this:

```cs
).Task.ForgetAndLogOnFailure();
```